python2/httplib2/socks.py

```diff
@@ -113,7 +113,7 @@ def wrapmodule(module):
     else:
         raise GeneralProxyError((4, "no proxy specified"))
 
-class socksocket(socket.socket):
+class socksocket(socket.socket().__class__):
```

Comment:
I'm not sure this is a good idea. How about to have the following code just after the `getattr` check?
```python
if hasattr(socket.socket.__class__, 'func_name'):
    # Jython's socket.socket is a function which wraps a class constructor.
    socket.socket = socket.socket().__class__
```